src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,125 @@
 package core.basesyntax;
 
+import java.util.Arrays;
+import java.util.NoSuchElementException;
+
 public class ArrayList<T> implements List<T> {
+    private static final double RESIZE_COUNT = 1.5;
+    private Object[] elementData;
+    private int defaultSize;
+    private int count;
+
+    public ArrayList() {
+        defaultSize = 10;
+        elementData = new Object[defaultSize];
+        count = 0;
+    }
+
+    private boolean rangeCheckForCount(int index) {
+        if (index >= count || index < 0) {
+            throw new ArrayListIndexOutOfBoundsException("Index out of bound of array.");
+        }
+        return true;
```

Comment:
You actually don't need to return anything from this method. Just make it void and use in all your methods as: rangeCheckForCount(index) without if. If something wrong with index - ok, exception will be throwed. With this approach you can delete unnecessary <return null;> statement from many methods in your code.